opentelemetry-api/src/opentelemetry/correlationcontext/propagation/__init__.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import re
+import typing
+import urllib.parse
+
+from opentelemetry import correlationcontext
+from opentelemetry.context import get_current
+from opentelemetry.context.context import Context
+from opentelemetry.trace.propagation import httptextformat
+
+
+class CorrelationContextPropagator(httptextformat.HTTPTextFormat):
+    _CORRELATION_CONTEXT_HEADER_NAME = "otcorrelationcontext"
+
+    def extract(
+        self,
+        get_from_carrier: httptextformat.Getter[
+            httptextformat.HTTPTextFormatT
+        ],
+        carrier: httptextformat.HTTPTextFormatT,
+        context: typing.Optional[Context] = None,
+    ) -> Context:
+        """ Extract CorrelationContext from the carrier.
+
+        See `opentelemetry.trace.propagation.httptextformat.HTTPTextFormat.extract`
+        """
+
+        if not context:
+            context = get_current()
+
+        header = _extract_first_element(
+            get_from_carrier(carrier, self._CORRELATION_CONTEXT_HEADER_NAME)
+        )
+
+        if not header:
+            return context
+
+        correlations = header.split(",")
+
+        for correlation in correlations:
+            try:
+                name, value = correlation.split("=", 1)
+            except Exception:  # pylint: disable=broad-except
+                continue
+            context = correlationcontext.set_correlation(
```

Comment:
It could be nice to have a function that receives a list (or dict) of key-value peers and adds them to the correlation context at once. It'll avoid to create that many temporal contexts that are overwritten in the next iteration of the loop.

Reply:
i agree, there was nothing in the spec for this but it could definitely be something we implement.